inject/src/main/java/io/micronaut/inject/ast/ClassElement.java

```diff
@@ -236,6 +382,27 @@ default List<PropertyElement> getBeanProperties() {
         return Collections.emptyList();
     }
 
+    /**
+     * Returns the native bean properties (synthetic getters and setters) for this class element..
+     *
+     * @return The bean properties for this class element
+     * @since 4.0.0
+     */
+    default List<PropertyElement> getNativeBeanProperties() {
+        return Collections.emptyList();
+    }
+
+    /**
+     * Returns the bean properties (getters and setters) for this class element based on custom configuration.
+     *
+     * @param configuration The configuration
+     * @return The bean properties for this class element
+     * @since 4.0.0
+     */
+    default List<PropertyElement> getBeanProperties(BeanPropertiesConfiguration configuration) {
```

Comment:
Add `@NonNull` to the return type and the argument. 

For consistency rename `BeanPropertiesQuery` as we already have `ElementQuery` so having this named with `Configuration` confuses matters